packages/desmos-std/src/profiles/mocks.rs

```diff
@@ -0,0 +1,408 @@
+                prefix: "cosmos".to_string()
+            },
+            proof: Proof {
+                pub_key: PubKey {
+                    proto_type: "/cosmos.crypto.secp256k1.PubKey".to_string(),
+                    key: "AyRUhKXAY6zOCjjFkPN78Q29sBKHjUx4VSZQ4HXh66IM".to_string()
+                },
+                signature: Signature {
+                    proto_type: "/desmos.profiles.v1beta1.SingleSignatureData".to_string(),
+                    signature: "C7xppu4C4S3dgeC9TVqhyGN1hbMnMbnmWgXQI2WE8t0oHIHhDTqXyZgzhNNYiBO7ulno3G8EXO3Ep5KMFngyFg".to_string()
+                },
+                plain_text: "636f736d6f733138786e6d6c7a71727172367a74353236706e637a786536357a6b33663478676d6e6470786e32".to_string()
+            },
+            chain_config: ChainConfig { name: "cosmos".to_string() },
+            creation_time: "2022-02-21T13:18:57.800827Z".to_string()
+        }
+    }
+
+    pub fn get_mock_application_link() -> ApplicationLink {
+        ApplicationLink {
+            user: Addr::unchecked("desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc"),
+            data: Data {
+                application: "twitter".to_string(),
+                username: "goldrake".to_string(),
+            },
+            state: "APPLICATION_LINK_STATE_VERIFICATION_SUCCESS".to_string(),
+            oracle_request: OracleRequest {
+                id: "537807".to_string(),
+                oracle_script_id: "32".to_string(),
+                call_data: CallData {
+                    application: "twitter".to_string(),
+                    call_data: "7b22757365726e616d65223a224c756361675f5f2335323337227d".to_string(),
+                },
+                client_id: "desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc-twitter-goldrake"
+                    .to_string(),
+            },
+            result: AppLinkResult::Success {
+                value: "4c756361675f5f2345423337".to_string(),
+                signature: "9690d734171298eb4cc9636c36d8507535264c1fdb136c9095a6a50c41ccffa"
+                    .to_string(),
+            },
+            creation_time: "2022-02-21T13:18:57.800827Z".to_string(),
+        }
+    }
+}
+
+/// Replacement for cosmwasm_std::testing::mock_dependencies
+/// this use our CustomQuerier to use desmos querier
+pub fn mock_dependencies_with_custom_querier(
+    contract_balance: &[Coin],
+) -> OwnedDeps<MockStorage, MockApi, MockQuerier<ProfilesQueryRouter>, ProfilesQueryRouter> {
+    let contract_addr = MOCK_CONTRACT_ADDR;
+    let custom_querier: MockQuerier<ProfilesQueryRouter> =
+        MockQuerier::new(&[(contract_addr, contract_balance)]).with_custom_handler(|query| {
+            SystemResult::Ok(ProfilesQueryRouter::custom_query_execute(query))
+        });
+    OwnedDeps::<_, _, _, ProfilesQueryRouter> {
+        storage: MockStorage::default(),
+        api: MockApi::default(),
+        querier: custom_querier,
+        custom_query_type: PhantomData,
+    }
+}
+
+pub trait MockQueries<T> {
```

Comment:
Since the instance which execute `query` function should be `Querier` instead of `QueryRouter`, it should be replaced by a `MockProfilesQuerier` like:
```rust
pub struct MockProfilesQuerier {}

impl MockSubspacesQuerier {
    fn custom_query_execute(query: &ProfilesQueryRouter) ->  ContractResult<Binary> { .. }
}
```